src/transaction/lock_manager.c

```diff
@@ -1077,6 +1079,33 @@ lock_initialize_tran_lock_table (void)
     }
 
   return NO_ERROR;
+
+exit_on_error:
+  if (lk_Gl.tran_lock_table == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1,
+	      (size_t) (SIZEOF_LK_TRAN_LOCK * lk_Gl.num_trans));
+
+      return ER_OUT_OF_VIRTUAL_MEMORY;
+    }
+
+  free_and_init (lk_Gl.tran_lock_table);
+  for (int k = 0; k <= i; k++)
+    {
+      tran_lock = &lk_Gl.tran_lock_table[k];
```

Comment:
1. lk_Gl.tran_lock_table is alread freeed in the 1092 line
2. The same behavior is in the lock_finalize() function. How about making this behavior a separate function?

Reply:
@hornetmj 

I changed the overlapping part to a separate function `lock_finalize_tran_lock_table ()`. It definitely looks cleaner.

Reply:
Different from the current code, I also thought about increasing the value of `lk_Gl.num_trans` for each loop, but I stopped because I was afraid I would use a lot of macros `MAX_NTRANS` in the code.
what do you think about it?